Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java

```diff
@@ -124,81 +269,73 @@ public boolean tryAddAnchor(ConnectionAnchor anchor, boolean overrideExisting, b
     }
 
     /**
-     * @param anchor
-     *            Anchor to get the slot of
-     * @return The slot (startAnchor or endAnchor variable) this anchor should
-     *         be put in if connected to this Connection
+     * Properly disconnects the given anchor from this Connection, notifying the anchor of its disconnection.
      */
-    private Optional<? extends ConnectionAnchor> getAnchorSlot(ConnectionAnchor anchor) {
-        if (anchor instanceof OutputAnchor) {
-            return startAnchor;
-        } else if (anchor instanceof InputAnchor) {
-            return endAnchor;
-        } else {
-            return Optional.empty();
+    public final void disconnect(ConnectionAnchor anchor) {
+        boolean disconnected = false;
+        if (startAnchor.isPresent() && startAnchor.get().equals(anchor)) {
+            startAnchor = Optional.empty();
+            disconnected = true;
+        } else if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
+            endAnchor = Optional.empty();
+            disconnected = true;
+        }
+        
+        if(disconnected) {
+            removeListeners(anchor);
+            anchor.disconnectConnection(this);
+            ConnectionCreationManager.nextConnectionState();
+    
+            //Let the now disconnected anchor update its visuals.
+            anchor.invalidateConnectionStateCascading();
+            //Let the remaining connected anchors update their visuals.
+            invalidateConnectionStateCascading();
         }
     }
 
+    /**
+     * Disconnects both anchors.
+     */
+    public final void disconnect() {
+        startAnchor.ifPresent(a -> disconnect(a));
+        endAnchor.ifPresent(a -> disconnect(a));
+    }
 
+    /**
+     * Removes this Connection, disconnecting its anchors and removing this Connection from the pane it is on.
+     */
+    public final void remove() {
+        disconnect();
+        getPane().getChildren().remove(this);
+    }
+
+    /**
+     * @return True if both anchors are present and their types match.
```

Comment:
Are you sure? I haven't tested it, but looking at the implementation one would think it would also return true if the anchors are not present/connected.